### PR TITLE
Cleanup ASM/mixins coremods

### PR DESCRIPTION
**Pull request by Alexdoru**
Cleanup asm transformers
Translate GT Pollution ASM transformers to mixins
Fix optifine support for GT Pollution mixins
Now require devs to set Phase and Side when registering mixins to avoid mistakes and crashes in production

All the pollutions related byecode manipulation stuff is now in mixins form and should be moved to GT5U in a later PR

### CODE REVIEW

**Comment by Glease** [comment]
does it still works with optifine? I've seen optifine something in the file history so it might be interfering.

**Comment by Alexdoru** [comment]
> does it still works with optifine? I've seen optifine something in the file history so it might be interfering.

nothing injects in the RenderBlocks class when optifine is present, I'll try to change it

**Comment by Alexdoru** [comment]
we work with optifine now 
image

File: src/main/java/com/mitchej123/hodgepodge/core/HodgepodgeCore.java
Comment (Caedis) [inline src/main/java/com/mitchej123/hodgepodge/core/HodgepodgeCore.java:1]: Should probably remove the commented out lines in this file but that is pretty minor

**Review by Caedis — APPROVED** [review]

**Comment by Caedis** [comment]
Please do not use & in branch names

**Comment by Alexdoru** [comment]
I rebased on top of master locally as well as squashed a bunch of commits together to clean the git log, I made sure it builds and runs

**Comment by Caedis** [comment]
I was double checking it just now lol

**Comment by Caedis** [comment]
Builds and launches without issue on my end btw.

**Pull request merged**